db/exec.c

```diff
@@ -44,7 +44,11 @@ u32 db_data_version_get(struct db *db)
 	struct db_stmt *stmt;
 	u32 version;
 	stmt = db_prepare_v2(db, SQL("SELECT intval FROM vars WHERE name = 'data_version'"));
-	db_query_prepared(stmt);
+	/* postgres will act upset if the table doesn't exist yet. */
+	if (!db_query_prepared(stmt)) {
+		tal_free(stmt);
```

Comment:
is the stmt free anyway at the end?

Reply:
Yes, but the issue here is that postgres will not complete the query if the table does not exist.  Instead we get something like:
```
ERROR:  relation "vars" does not exist at character 20
STATEMENT:  SELECT intval FROM vars WHERE name = 'data_version'
```
That means db_stmt.row is left uninitialized, so it's not safe to call db_step.  Sqlite3 isn't so picky and will return an empty result, but we also need to handle the case of a new database under postgres.